packages/cascara/src/ui/Dashboard/Dashboard.js

```diff
@@ -50,32 +50,35 @@ const Dashboard = ({ as = 'div', config, ...rest }) => {
     const key = rest?.title + widget || index;
     const Component = WIDGETS[widget];
 
-    return Component ? (
-      <Component {...rest} key={key} />
-    ) : (
-      // TODO: This should eventually become a message component for displaying helpful developer messages
-      <WidgetError
-        key={key}
-        message={
-          <>
-            <p>
-              <code>{widget}</code>
-              {' is not a valid value for '}
-              <code>widget</code>.
-            </p>
-            <p>
-              Try:{' '}
-              {widgetKeys.map((key, i) => (
-                <>
-                  <code key={key}>{key}</code>
-                  {widgetKeys.length > i + 1 && ', '}
-                </>
-              ))}
-            </p>
-          </>
-        }
-      />
-    );
+    // render null if no data
+    return rest.data && rest.data.length > 0 ? (
+      Component ? (
+        <Component {...rest} key={key} />
+      ) : (
+        // TODO: This should eventually become a message component for displaying helpful developer messages
+        <WidgetError
+          key={key}
+          message={
+            <>
+              <p>
+                <code>{widget}</code>
+                {' is not a valid value for '}
+                <code>widget</code>.
+              </p>
+              <p>
+                Try:{' '}
+                {widgetKeys.map((key, i) => (
+                  <>
+                    <code key={key}>{key}</code>
+                    {widgetKeys.length > i + 1 && ', '}
+                  </>
+                ))}
+              </p>
+            </>
+          }
+        />
+      )
+    ) : null;
```

Comment:
@lordpixel we should make a distinction between undefined or empty data. Undefined data in Cascara should show as a loader. Empty data should show as an empty state.

Technically, the way the widgets are built for Dashboard, we should be able to go to the Widget component that has the title and shared widget logic and place a wrapper to render a loading state if there is no data to pass to any widget. 

I think returning `null` from any of our components is almost always going to be the wrong approach. From our system components, we probably never want to return nothing. If something should not render at all, that logic should exist out of Cascara. Otherwise, devs should be able to pass data as it comes in, and if that data happens to be undefined at the time of render, that means loading.

Reply:
@brianespinosa I agee with you on the fact that rerturning null is not the correct approach. 

Checking `Widget` component, I can see we are already using css classes for undefined/empty data. Should that be ennough? please see [Widget.js](https://github.com/Espressive/cascara/blob/develop/packages/cascara/src/ui/Dashboard/widgets/Widget.js#L75)

Reply:
@lordpixel it seems that based on the updates to Widget.js that this bug report might not be valid anymore. That was what I was trying to ask on the ticket when I was wondering if this is still valid. Are we able to reproduce the reported problem anymore? If not, we can probably go ahead and decline this PR. If the problem still presents itself, I think there is a different issue going on.